python/plugins/processing/algs/qgis/Datasources2Vrt.py

```diff
@@ -82,151 +81,44 @@ def defaultFileExtension(self):
         self.addOutput(QgsProcessingOutputString(self.VRT_STRING,
                                                  self.tr('Virtual string')))
 
-    def name(self):
-        return 'buildvirtualvector'
-
-    def displayName(self):
-        return self.tr('Build virtual vector')
-
     def processAlgorithm(self, parameters, context, feedback):
         input_layers = self.parameterAsLayerList(parameters, self.INPUT, context)
         unioned = self.parameterAsBoolean(parameters, self.UNIONED, context)
         vrtPath = self.parameterAsOutputLayer(parameters, self.OUTPUT, context)
 
-        vrtString = self.mergeDataSources2Vrt(input_layers,
-                                              vrtPath,
-                                              union=unioned,
-                                              relative=False,
-                                              schema=False,
-                                              feedback=feedback)
-        return {self.OUTPUT: vrtPath, self.VRT_STRING: vrtString}
-
-    def mergeDataSources2Vrt(self, dataSources, outFile, union=False, relative=False,
-                             schema=False, feedback=None):
-        '''Function to do the work of merging datasources in a single vrt format
-
-        @param data_sources: Array of path strings
-        @param outfile: the output vrt file to generate
-        @param relative: Write relative flag. DOES NOT relativise paths. They have to be already relative
-        @param schema: Schema flag
-        @return: vrt in string format
-        '''
-        if feedback is None:
-            feedback = QgsProcessingFeedback()
-
         vrt = '<OGRVRTDataSource>'
-        if union:
+        if unioned:
             vrt += '<OGRVRTUnionLayer name="UnionedLayer">'
 
-        total = 100.0 / len(dataSources) if dataSources else 1
-        for current, layer in enumerate(dataSources):
+        total = 100.0 / len(input_layers) if input_layers else 0
+        for current, layer in enumerate(input_layers):
             if feedback.isCanceled():
                 break
 
+            uri = layer.source()
```

Comment:
Shouldn't this use QgsProviderRegistry.instance().decodeUri instead?

Reply:
I think better to use GDAL connection string here, to allow also database and WFS layers. Should be fixed now.

Reply:
Good move!